src/library/dao/analysisdao.cpp

```diff
@@ -323,14 +323,26 @@ void AnalysisDao::saveTrackAnalyses(TrackInfoObject* pTrack) {
     ConstWaveformPointer pWaveform = pTrack->getWaveform();
     ConstWaveformPointer pWaveSummary = pTrack->getWaveformSummary();
 
+    TrackId trackId(pTrack->getId());
+
+    // Delete waveform analysis if track was requested to have its waveform cleared.
+    if (pTrack->isClearWaveformRequested()) {
+        bool success = deleteAnalysesForTrack(trackId);
+        qDebug() << (success ? "Successfully deleted" : "Failed to delete")
+                 << "waveform analysis for trackId" << trackId;
+        if (success) {
```

Comment:
If you don't reset the flag in case deleteAnalysesForTrack(trackId) fails you will repeatedly enter this code path without actually resetting the flag. Maybe the analysis has already been deleted? I can't think of any other reason why deletion should fail. You might log a warning with qWarning() on failure, but the TIO's flag should always be reset. The proposed function should return true if the flag has been reset and false otherwise.